lib/signature.dart

```diff
@@ -196,17 +201,88 @@ class SignatureController extends ValueNotifier<List<Point>> {
     }
     _SignaturePainter(this).paint(canvas, null);
     var picture = recorder.endRecording();
-    return picture.toImage(
-        (maxX - minX + penStrokeWidth * 2).toInt(), (maxY - minY + penStrokeWidth * 2).toInt());
+    return picture.toImage((maxX - minX + penStrokeWidth * 2).toInt(),
+        (maxY - minY + penStrokeWidth * 2).toInt());
   }
 
   Future<Uint8List> toPngBytes() async {
-    var image = await toImage();
-    if (image == null) {
-      return null;
+    try {
+      var image = await toImage();
+      if (image == null) {
+        return null;
+      }
+
+      var bytes = await image.toByteData(format: ui.ImageByteFormat.png);
+      return bytes.buffer.asUint8List();
+    } catch (e) {
+      return _toPngBytesForWeb();
     }
+  }
+
+  //'image.toByteData' is not available for web. So we are use the package
+  // "image" to create a image which works on web too
+  Uint8List _toPngBytesForWeb() {
```

Comment:
It would be interesting to have one implementation for all platforms. Since image package is already added with this PR, it may be reasonable to try to use it also for mobile devices.

Or is there something that would prevent it?

Reply:
The actual exported results on mobile/macos are better than the ones on web. My web implementation for the image export is not as "smooth" as your native implementation. Maybe it can be improved with some fine tuning in the code. For me it's enough since web is still not popular with flutter and signatures usually aren't done on a website... But I agree a universal implementation would be more interesting

Reply:
OK we will leave it as two separate implementations for now :)